### PR TITLE
(554545) Remove double logging

### PR DESCRIPTION
**Pull request by MarkZuber**
Please pay special attention to changes made in LoggerCallbackTests.cs and AdalLoggerBase::Log() to ensure behavior is consistent with legacy.

### CODE REVIEW

**Comment by bgavrilMS** [comment]
While having a look at the MSALLogger, I noticed that one of the methods throws an exception - is this expected? If not, probably missing a unit test as well.. Please check ADALLogger as well. Otherwise the PR looks good!

 public override void InfoPii(string messageWithPii, string messageScrubbed)
        {
            throw new NotImplementedException();
        }

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationParameters.cs [hunk 1]
```diff
@@ -163,8 +161,6 @@ private static async Task<AuthenticationParameters> CreateFromResourceUrlCommonA
                 if (response == null)
                 {
                     var serviceEx = new AdalServiceException(AdalErrorMessage.UnauthorizedHttpStatusCodeExpected, ex);
-                    string noPiiMsg = CoreExceptionFactory.Instance.GetPiiScrubbedDetails(serviceEx);
-                    CoreLoggerBase.Default.Error(noPiiMsg);
```
Comment (bgavrilMS) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationParameters.cs:167]: >noPiiMsg [](start = 49, length = 8)

Tedious work, thanks for doing this! #ByDesign

**Review by bgavrilMS — APPROVED** [review]
:shipit:

**Comment by MarkZuber** [comment]
>           While having a look at the MSALLogger, I noticed that one of the methods throws an exception - is this expected? If not, probably missing a unit test as well.. Please check ADALLogger as well. Otherwise the PR looks good!
> public override void InfoPii(string messageWithPii, string messageScrubbed)
> {
> throw new NotImplementedException();
> }
You may have been looking at an intermediary commit.  I've re-reviewed the code and I don't see any NotImplementedExceptions being thrown within any of the loggers.

**Pull request merged**